packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1502,15 +1502,20 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     final MediaQueryData mediaQuery = MediaQuery.of(context);
     final ThemeData themeData = Theme.of(context);
     final TextDirection textDirection = Directionality.of(context);
+    final bool semanticsEnabled = RendererBinding.instance.pipelineOwner.semanticsOwner != null;
```

Comment:
This might have unintended consequences. Some apps on Android enable accessibility services (and thus semantics) to remote control the app (password managers, etc). 

We should double-check how the native implementations handle this. Somebody once told me that they check if explore-on-touch is enabled and only then disable the timeout. We might have to introduce a new flag besides semantics.

Reply:
From what Ive been able to find, it looks like the example implementation's timer is paused whenever there are touch events happening, maybe this is enough

https://github.com/material-components/material-components-android/blob/7aff9d2e8b5d62d3d7c554d3debfe9f1a11a7c77/lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java#L762

Reply:
and also if there are swipes happening

https://github.com/material-components/material-components-android/blob/7aff9d2e8b5d62d3d7c554d3debfe9f1a11a7c77/lib/java/com/google/android/material/snackbar/BaseTransientBottomBar.java#L450

Reply:
It looks like the accepted behavior is to use isTouchExplorationEnabled.